File: app/src/main/java/ru/magflayer/spectrum/domain/interactor/FileManagerInteractor.java
package ru.magflayer.spectrum.domain.interactor;

import android.net.Uri;

import javax.inject.Inject;

import ru.magflayer.spectrum.domain.repository.FileManagerRepository;
import rx.Observable;

public class FileManagerInteractor {

    private final FileManagerRepository fileManagerRepository;

    @Inject
    public FileManagerInteractor(final FileManagerRepository fileManagerRepository) {
        this.fileManagerRepository = fileManagerRepository;
    }

    public Observable<Uri> saveFileToExternalStorage(final String fileName, final byte[] bytes) {
        return fileManagerRepository.saveFileToExternalStorage(fileName, bytes);
    }
}

File: app/src/main/java/ru/magflayer/spectrum/domain/interactor/FileManagerInteractor.kt
package ru.magflayer.spectrum.domain.interactor

import android.net.Uri
import ru.magflayer.spectrum.domain.repository.FileManagerRepository
import javax.inject.Inject

class FileManagerInteractor @Inject constructor(
    private val fileManagerRepository: FileManagerRepository,
) {

    suspend fun saveFileToExternalStorage(fileName: String, bytes: ByteArray): Uri {
        return fileManagerRepository.saveFileToExternalStorage(fileName, bytes)
    }

    suspend fun copyToLocalStorage(sourceFilePath: Uri): Uri {
        return fileManagerRepository.copyFileToExternalEndpoint(sourceFilePath)
    }
}
